docker/Dockerfile-depl

```diff
@@ -3,7 +3,8 @@ FROM ubuntu:focal-20220113 AS install_dependencies
 ENV DEBIAN_FRONTEND=noninteractive
 RUN apt-get update  && apt-get install -y cmake git build-essential libgtest-dev libssl-dev qtbase5-dev \
     zip libmysqlcppconn-dev libboost1.71-all-dev libmysqlclient-dev uuid-dev zlib1g libxerces-c-dev qtbase5-dev \
-    libcurl4-openssl-dev libgps-dev libsnmp-dev librdkafka-dev libjsoncpp-dev libev-dev libuv1-dev libcpprest-dev
+    libcurl4-openssl-dev libgps-dev libsnmp-dev librdkafka-dev libjsoncpp-dev libev-dev libuv1-dev libcpprest-dev \
+    curl
```

Comment:
Is this curl dependency needed for this change?

Reply:
@dan-du-car  and @paulbourelly999 
With the latest changes that Cody and I have merged we have removed the separate Dockerfile-depl.  I have merged them into this branch to ensure there is no merge conflict and so this change is overcome by events.  It did not seem that curl was needed but if it is we can put it in the correct place.

Reply:
Yes curl is needed for the calls in the ImmediateForward and MessageReceiver plugins. When you say you removed the Dockerfile-depl does that mean you replaced the existing one with the Dockerfile-depl. I created this docker file to remove unused dependencies if I remember correctly.

Reply:
@paulbourelly999 The process for Docker has been modified.  Previously the Dockerfile was used in the Sonar Build and for x64 buils while the Dockerfile-depl was used for only arm builds.  Cody and I figured out that we did not need Sonar Scanner to use a Docker image with everything in it so we have moved some of the commands that were in the Dockerfiles (there was a  lot of duplication) into scripts as these scripts can also be used for developers working directly on a VM.   Either curl can be added in the install_dependencies.sh script or this can be reworked to use libcurl like is done in the CARMA Cloud plugin.  Probably using the library is better than making a system call.